### PR TITLE
Migrate custom selects (Chakra -> Ant)

### PR DESCRIPTION
**Pull request by gilluminate**
Closes HJ-160

### Description Of Changes

* Migrate remaining Select input fields from Chakra to Ant Design.
* Replace disabled "creatable" select instances where we were using them just for badge display. Use pure Ant Badges instead.

### Code Changes

* Cleaned up a few files/components that are obsolete that were using old Chakra Select components
* Changed `colorPrimaryBg` to `palette.FIDESUI_NEUTRAL_75` as per @jack-gale-ethyca's request
* Migrated one more "one-off" in the System Inventory page.
* Replace disabled Select instances with Ant badges

### Steps to Confirm

<hr/>

1. Visit System inventory where an integration has been configured (Iterable)
2. Switch to the "Integrations" tab
3. Ensure the "Request types" multi-select uses the new Select
4. Now switch to the "Data uses" tab
5. Click one of the options or the [Add data use +] button
6. In the modal, the "Data use," "Data categories," "Data subjects," "Legal basis for processing," "Features," "Dataset references," and "Cookies" fields all use the new Select 

<hr/>

1. Visit the Manage your vendors page (`/consent/configure`)
2. Click [Add Vendors] and then [Add custom vendor]
3. All fields in this modal are using the new Select
7. Close the modal and return to the Manage your vendors page
8. open a Vendor that has more than one TCF Purpose
9. Badges should appear here where there used to be a disabled Select instance with badges inside it.

CleanShot 2024-11-14 at 15 15 55@2x

<hr/>

1. Visit Manage custom fields (`/settings/custom-fields`)
2. Add a new custom field
10. "Location" and "Field type" are both using the new Select.
11. Go ahead and create 1 for Single Select
12. Go to the Add system form and at the very bottom under Custom fields you should see the custom field you just created and it should be using the new Select.

<hr/>

1. Go to Add Systems (`/add-systems`)
4. Click the "Add a System" button
5. The "Dataset references" field is using the new Select (everything else on this page was updated in the previous PR)
3. Now on the Add Systems page, Click the "Scan your infrastructure (AWS)" button
4. The AWS Region field is using the new Select

<hr/>

1. Go to add New Privacy Notice (`/consent/privacy-notices/new`)
2. "Consent mechanism," "Data use," and "Enforcement level" are all using the new Select

<hr />

1. Go to Integrations (`/integrations`)
2. Click [Add Integration] button
3. Click [Configure] button on one of the options
7. The "System" field is using the new Select
8. Close the modal and [Manage] one of the existing integrations (or crate one on the previous step)
9. Click the "Data discovery" tab
13. Click the [Add monitor] button
14. the "Automatic execution frequency" field is using the new Select

<hr />

1. Go to Organizations Management (`/settings/organization`)
2. Click [Add SSO Provider]
3. The "Provider" field is using the new Select

<hr />

1. Go to create a new Experience (`/consent/privacy-experience/new`)
2. The "Experience Type" field is using the new Select
3. Use that to select a "Banner and modal" type
5. The now visible "Banner options" field uses the new Select

<hr />

1. Go to Privacy requests (`/privacy-requests`)
2. Click [Create request]
3. The "Request type" field is using the new Select
4. Click the [filter] button at the top of the table
5. The "Status" and "Request type" fields are using the new Select

<hr />

1. Go to the (hidden) Configure Storage page (`/privacy-requests/configure/storage`)
2. "Format," and "Auth method" are using the new Select

<hr />

1. Go to Properties (`/properties`)
2. Click [Add Property] button
3. The "Type" field is using the new Select

<hr />

1. Go to the Data map report (`/reporting/datamap`)
2. Click a row to open the left Drawer
3. Expand an accordion under the "Data uses" section
6. There are 4 fields here using the new Select (these may or may not be disabled)

<hr />

1. Go to Taxonomy management (`/taxonomy`)
2. Switch to the Data Subjects tab
3. Click the [Add Taxonomy Entity] button
4. the "Rights" field is using the new Select

<hr />

1. Go to Connection manager (AKA Unlinked connections `/datastore-connection`)
2. Click the [Create new connection] button
3. The "Show all" connector type filter uses the new Select
4. Search for "Manual"
7. Click the "Manual Process" option
8. Give it a name and click Save
9. the "Data Categories" field uses the new Select

<hr />

### Pre-Merge Checklist

* [x] Issue requirements met
* [ ] All CI pipelines succeeded
* [x] `CHANGELOG.md` updated
* Followup issues:
  * [x] Followup issues created: [HJ-252](https://ethyca.atlassian.net/browse/HJ-252)
  * [ ] No followup issues
* Database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!
  * [x] No migrations
* Documentation:
  * [ ] Documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] Documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] If there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
  * [x] No documentation updates required

[HJ-252]: https://ethyca.atlassian.net/browse/HJ-252?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jpople — COMMENTED** [review]
Code looks good-- UAT'd as described in PR and confirming everything works.

I am seeing one bug on the connection manager page where the "Connection type" filter select doesn't render options, not sure if that was introduced with this PR or missed on the earlier one:

Screenshot 2024-11-20 at 13 30 20

Also, after clicking "add connection", this ConnectionTypeFilter component appears to still be using the old, bespoke SelectDropdown component, could we switch that over to Ant as well?  Its counterpart MultiSelectDropdown is also still in the codebase but doesn't seem to be used anywhere anywhere and should be able to be removed.

Screenshot 2024-11-20 at 13 33 20

File: clients/admin-ui/src/features/common/nav/v2/BackButton.tsx [hunk 1]
```diff
@@ -14,20 +15,23 @@ import NextLink from "next/link";
...
-      Back
-    </Text>
-  </Flex>
-);
+}: { backPath: string } & FlexProps) => {
+  const nextRouter = useRouter();
+  return (
+    <Flex alignItems="center" mb={6} {...props}>
+      <Button
+        onClick={() => nextRouter.push(backPath)}
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/common/nav/v2/BackButton.tsx:23]: Good question, I meant to leave a comment on this PR about it and forgot...

This was more of an annoyance fix. The link part would work as expected in a SPA where the page would immediately load. This button part however would load the entire app again. This change just aligns the button behavior with the link behavior using the Next router.

**Comment by gilluminate** [comment]
@jpople Thanks for finding that one-off select. It helped me find 2 more in the Privacy Requests table filter!

**Review by jpople — APPROVED** [review]
Looking good!  Thanks for getting those extra old selects, it's great to have that cleanup done.

**Pull request merged**